clarification for ResourceQuota concept (#42685)
kubernetes/website: content/en/docs/concepts/policy/resource-quotas.md

## Object Count Quota

You can set quota for **the total number of one particular resource** of all standard, namespaced resource types 

> using the following syntax:

Reviewer: My experiments showed that the `<resource>` in `count/<resource>` can be anything.
It can be something that doesn't exist at all.

Reviewer: So ... this doen't have to be a "standard" resource type, right?

Reviewer 2: Although the API accepts any resource name in that field, this could mean:
1. you can set any ResourceQuota, even for an API the cluster doesn't yet have
   - if the cluster gains that resource kind, the quota starts to be enforced
1.  you can set any ResourceQuota, even for an API the cluster doesn't yet have
    - however, there is an internal list of API kinds where enforcement happens
    - if you set a ResourceQuota for an different resource kind, the ResourceQuota is not enforced, and the lack of enforcement is silent

Let's be clear on which of those two cases applies.

Reviewer: I think the 1st case is the correct answer, but has to double confirm this.

Reviewer 3: It's case 2 (very well considered).  Though the "default" aggregated apiserver (APIService) enforces quota.

Reviewer 4: 'standard' is indeed ambiguous, better to clearly list out: built-in API and CRD

> for certain resources of all standard, namespaced resource types using the following syntax:

Author: Indeed it brings more confusion. I switch to another expression to see if it is better.

> The same syntax can be used for custom resources.

Reviewer 2: ```suggestion
If you define a quota this way, it applies to Kubernetes' APIs that are part of the API server, and
to any custom resources backed by a CustomResourceDefinition. If you use [API aggregation](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/) to
add additional, custom APIs that are not defined as CustomResourceDefinitions, the core Kubernetes
control plane does not enforce quota for the aggregated API. The extension API  server is expected to
provide quota enforcement if that's appropriate for the custom API.
```

For example, to create a quota on a `widgets` custom resource in the `example.com` API group, use `count/widgets.example.com`.

> When using such a resource quota (only available for some object kinds), an object is charged

Author: Sure. Correct this one to "nearly for all resources".

> against the quota if the control plane has a stored copy.

Reviewer 4: "if the control plane has a stored copy." is not very clear what this means, may be something like: "if the object kind exists (is defined) in the control plane"

The same syntax can be used for custom resources.
For example, to create a quota on a `widgets` custom resource in the `example.com` API group, use `count/widgets.example.com`.

> When using `count/*` resource quota for certain resources type, an object is charged against the quota if it exists in server storage.

Reviewer 2: ```suggestion
When using `count/*` resource quota (only available for some object kinds), an object is charged
against the quota if the control plane has a stored copy.
```
?

Author: Looks better! Thanks!

quota on a namespace to avoid the case where a user creates many small pods and
exhausts the cluster's supply of Pod IPs.

> You can find more examples on [Viewing and Setting Quotas](/docs/concepts/policy/resource-quotas/#viewing-and-setting-quotas).

Reviewer 2: ```suggestion
You can find more examples on [Viewing and Setting Quotas](#viewing-and-setting-quotas).
```
